spec/oas_v1.yaml

```diff
@@ -0,0 +1,497 @@
+          description: Numeric id of the proposal type
+          type: integer
+        # Note: quorum and threshold are strings since I'm not sure about the unit/size here 
+        # (i.e. if they can be contained in an int32/int64 type)
+        quorum:
+          summary: Minimmum participation for passage
+          description: >
+            The minimum number of voting power needed to be involved in a given proposal as a prerequisite for passage
+          type: string 
+        approvalThreshold:
+          summary: Amount needed for measure to pass 
+          description: The amount of voting power needed to pass the given proposal.
+          type: string
+    ExtendedProposalType:
+      allOf:
+        - $ref: "#/components/schemas/BaseProposalType"
+        - summary: Metadata about the proposal
+          description: An object describing the particular voting parameters of a proposal.
+          type: object
+          properties:
+            contractAddress:
+              summary: Governor contract address
+              description: Governor contract address to which this proposal was submitted.
+              type: string
+            createBlock:
+              summary: Proposal creation block number
+              description: Block number on which this proposal type was created.
+              type: string
+    VoteSupport:
+      summary: Disposition of the vote
+      description: Describes which way the vote was cast (i.e. for/against/abstention)
+      type: string
+      enum: 
+        - FOR 
+        - AGAINST
+        - ABSTENTION
+    Vote:
+      summary: Information about a given vote
+      description: Information and metadata about a specific vote on a governance proposal.
+      type: object
+      properties:
+        transactionId:
+          summary: Transaction ID of the vote
+          description: The transaction ID of the transaction in which the vote was cast.
+          type: string
+        proposalId: 
+          summary: Proposal ID for the vote
+          description: The unique ID of the proposal on which the vote was cast.
+          type: string
+        voterAddress:
+          summary: Address casting the vote
+          description: On-chain address of the EOA or contract which cast the vote.
+          type: string
+        support:
+          $ref: "#/components/schemas/VoteSupport"
+        weight:
+          summary: Voting power behind the vote
+          description: Numeric description of the weight and voting power behind the vote.
+          type: string
+        castBlock:
+          summary: Block in which the vote was cast
+          description: The block number in which the transaction for the cast is contained.
+          type: string 
+        description:
+          summary: Rationale for the vote
+          description: A voter-supplied reason for voting the particular way they did.
+          type: string
+        # Clarify intent
+        approvalParams:
```

Comment:
This will be used for approval votes which are basically multiple choice votes. We can include array of params here for simplicity